tools/performance-benchmarking/src/md/0010-content.md

```diff
@@ -0,0 +1,916 @@
+```
+
+</div>
+<div class="col col-33">
+
+This says: run two *variants* (`optimized` and `release`) of a single benchmark *suite* (`all_cbmc_proofs`).
+The variants are distinguished in this case by the `CBMC` environment variable, which the AWS [proof build system](https://github.com/model-checking/cbmc-starter-kit) uses when invoking CBMC.
+
+</div>
+</div>
+
+`benchcomp` copies the `all_cbmc_proofs` directory to two temporary directories, one for each variant, and runs the command line. It uses the built-in `litani_to_benchcomp` parser to assemble the results. `benchcomp` then writes this data to the output file in JSON format (here in YAML for readability):
+
+<div class="twocolumn">
+<div class="col col-75">
+
+```yaml
+# result.json
+
+metrics:
+  runtime:
+    lower_is_better: true
+    unit: s
+  passed: {}
+  coverage:
+    unit: %
+benchmarks:
+  s2n_init:
+    tags: [s2n, crypto, cbmc]
+    variants:
+        optimized:
+          metrics:
+            runtime: 3
+            passed: true
+            coverage: 100
+          path: ./tmp/benchcomp/runs/abcd1234/optimized/all_cbmc_proofs/s2n/tests/cbmc/proofs/s2n_init
+        release:
+          metrics:
+            runtime: 4
+            passed: true
+            coverage: 100
+          path: ./tmp/benchcomp/runs/abcd1234/release/all_cbmc_proofs/s2n/tests/cbmc/proofs/s2n_init
+  freertos_list_replace:
+    tags: [cbmc, freertos, freertos_kernel]
+    variants:
+      optimized:
+        metrics:
+          # ...
+```
+</div>
+<div class="col col-25">
+
+The output file first declares the various *metrics* that the union of all the parsers emitted.
+In later sections, we will see how Alice writes her own parser or extends an existing one with new metrics.
+The file then maps benchmark names to values for each of the metrics for each of the variants that the benchmark ran under.
+
+</div>
+</div>
+
+## Visualizing the results
+
+Alice now wants to visualize the difference in runtime between the two versions of CBMC, ignoring other metrics for now.
+Alice adds the `visualize` top-level key to her `benchcomp.yaml` file:
+
+<div class="twocolumn">
+<div class="col col-66">
+
+```yaml
+# add to benchcomp.yaml
+
+visualize:
+  - type: dashboard
+    output: comparison.html
+    graphs:
+      - type: pairwise_box_whisker
+        metric: runtime
+        pairs:
+          - [release, optimized]
+```
+
+</div>
+<div class="col col-33">
+
+This says to generate a single graph in a single HTML dashboard visualization.
+The graph compares the runtimes of all benchmarks run under the `release` variant pairwise with the same benchmarks run under the `optimized` variant.
+
+</div>
+</div>
+
+Alice runs the entire suite, together with generating and writing out the visualization, by running `benchcomp` again.
+Alternatively, she can run `benchcomp visualize < result.json`, which loads the result of the run she did in the previous section.
+
+The resulting dashboard looks like this:
```

Comment:
I have a few questions about this graph, not sure it's out of the scope of this PR.